sig-release/members.md

```diff
@@ -0,0 +1,8 @@
+## SIG Release Membership
```

Comment:
I would recommend moving this into README.md as "Organizers" and use inclusion in the kubernetes-sig-release google group(s) as membership, rather than having to update this table